operators/multiclusterobservability/pkg/config/config.go

```diff
@@ -482,6 +482,18 @@ func GetDefaultTenantName() string {
 
 // GetObsAPIHost is used to get the URL for observartium api gateway.
 func GetObsAPIHost(client client.Client, namespace string) (string, error) {
+	mco := &observabilityv1beta2.MultiClusterObservability{}
+	err := client.Get(context.TODO(),
+		types.NamespacedName{
+			Name: GetMonitoringCRName(),
+		}, mco)
+	if err != nil && !errors.IsNotFound(err) {
+		return "", err
+	}
+	advancedConfig := mco.Spec.AdvancedConfig
+	if advancedConfig != nil && advancedConfig.CustomObservabilityHubURL != "" {
+		return string(advancedConfig.CustomObservabilityHubURL), nil
```

Comment:
I know there is a kubebuilder annotation for the regex but would it be worth calling https://pkg.go.dev/net/url#ParseRequestURI here and returning an error if we hit one?